### PR TITLE
Add azure OPENAI support

### PR DESCRIPTION
**Pull request by sykuang**
Thank you for creating such a helpful extension. I am using Microsoft Azure OpenAI, but I know only a few people will use Azure OpenAI, so I modified a little part to make it support Azure OpenAI. Since I am a novice to typescript, If you feel this implementation could be better, please reject the PR or let me know where I should improve.

### CODE REVIEW

**Comment by abielzulio** [comment]
Hi! @sykuang 

Could you provide also how to use Azure's OpenAI in the `README`? That would be much more helpful. Thanks!

File: package-lock.json [hunk 1]
```diff
@@ -2427,6 +2436,15 @@
         "follow-redirects": "^1.14.8"
       }
     },
+    "azure-openai": {
```
Comment (abielzulio) [inline package-lock.json]: Why do we need to install the dependency but ended up not using it? Am I missing something? or perhaps I need an explanation of how we are going to use it.

**Review by abielzulio — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (sykuang) [reply package-lock.json]: Sorry about this; I have removed it and pushed a new commit.

**Comment by sykuang** [comment]
> 

I have added some tips and screenshots to explain using Azure OpenAI. Feel free to let me know if you have any suggestions.

**Comment by abielzulio** [comment]
Closing it due now is already on review. Thank you for your contribution! @sykuang

**Pull request closed without merging**